client/cli/src/config.rs

```diff
@@ -98,15 +116,21 @@ pub trait CliConfiguration: Sized {
 	/// Gets the role
 	///
 	/// By default this is `Role::Full`.
-	fn role(&self, _is_dev: bool) -> Result<Role> {
-		Ok(Role::Full)
+	fn role(&self, is_dev: bool) -> Result<Role> {
+		Ok(self
+			.node_params()
+			.map(|x| x.role(is_dev))
+			.unwrap_or(Role::Full))
 	}
 
 	/// Get the transaction pool options
 	///
 	/// By default this is `TransactionPoolOptions::default()`.
 	fn transaction_pool(&self) -> Result<TransactionPoolOptions> {
```

Comment:
What is the connection between this and the `transaction_pool_params` and why both exists?

Reply:
TransactionPoolOptions is the configuration object from sc-transaction-pool, this is really the actual configuration for the transaction pool.

TransactionPoolParams is the structopt that's use only to parse the command line arguments and be transformed later on Into TransactionPoolOptions.

Both can be made to one single struct sometimes. But it is not always possible (structopt may limits you). And it might not be advisable to do so because you would need to add structopt to the dependencies of sc-transaction-pool while it should actually be CLI agnostic.